morphia/src/main/java/org/mongodb/morphia/dao/BasicDAO.java

```diff
@@ -24,17 +25,19 @@
  * @author Olafur Gauti Gudmundsson
  * @author Scott Hernandez
  */
-@SuppressWarnings("deprecation")
+@SuppressWarnings({"WeakerAccess", "deprecation", "unused"})
```

Comment:
Could you explain to me the new annotations and the causes?  I'm unfamiliar with "WeakerAccess" and haven't come across it before.